dashboard/app/models/user.rb

```diff
@@ -2342,8 +2350,51 @@ def increment_section_attempts
     save! if persisted?
   end
 
+  # The data returned by this method is set to cookies for the marketing team to
+  # use in Optimizely for segmenting teacher user experience.
+  def marketing_segment_data
+    return unless teacher?
+
+    {
+      locale: read_attribute(:locale),
+      account_age_in_years: account_age_in_years,
+      grades: grades_being_taught.any? ? grades_being_taught.to_json : nil,
+      courses: courses_being_taught.any? ? courses_being_taught.to_json : nil,
+      has_attended_pd: has_attended_pd?,
+      within_us: within_united_states?,
+      school_percent_frl: school_stats&.frl_eligible_total,
+      school_title_i: school_stats&.title_i_status
+    }.compact
+  end
+
+  def self.marketing_segment_data_keys
+    %w(locale account_age_in_years grades courses has_attended_pd within_usschool_percent_frl school_title_i)
+  end
+
   private
 
+  def account_age_in_years
+    ((Time.now - created_at.to_time) / 1.year).floor
+  end
+
+  # Returns a list of all grades that the teacher currently has sections for
+  def grades_being_taught
+    sections.map(&:grade).uniq
+  end
+
+  # Returns a list of all courses that the teacher currently has sections for
+  def courses_being_taught
```

Comment:
Should we call this something like `curriculums_being_taught` or something similar?  (I'm worried that `course` means something specific in the model.)